crates/ruff_python_ast/src/nodes.rs

```diff
@@ -179,7 +179,7 @@ pub struct StmtFunctionDef {
     pub is_async: bool,
     pub decorator_list: Vec<Decorator>,
     pub name: Identifier,
-    pub type_params: Option<TypeParams>,
+    pub type_params: Option<Box<TypeParams>>,
```

Comment:
What's the motivation for boxing some fields? it seems unrelated to the PR itself.